[STT-1363] - Fix HTML tags rendering in Assignment view

eos87 opened the pull request:
### Purpose

This PR fixes  the rendering of HTML tags in assignment descriptions and improving the assignment preview functionality. The changes ensure that HTML content in assignment descriptions is properly sanitized to display as plain text instead of rendering raw HTML tags, while also enhancing the assignment preview system with better coverage profile handling.

### What has changed

**HTML Sanitization:**
- Modified `DescriptionTextComponent` in `client/components/Assignments/AssignmentItem/fields/DescriptionText.tsx` to convert HTML content to plain text using `stringUtils.convertHtmlToPlainText()`, preventing raw HTML from being displayed in assignment descriptions

**Assignment Coverage Profile Enhancement:**
- Added new selector `getCurrentAssignmentCoverageProfile` in `client/selectors/assignments.ts` to properly retrieve coverage profiles used when creating assignments, including user modifications
- Updated `AssignmentPreview.tsx` and `AssignmentPreviewContainer/index.tsx` to utilize the new coverage profile selector
- Added tests in `client/selectors/tests/assignments_test.ts` to verify the new selector functionality

### How to test
Please follow the steps in the ticket 

### Screenshots
#### before
<img width="1000" alt="image" src="https://github.com/user-attachments/assets/ed7d326f-f13d-4118-acb1-c18bc6e76085" />

#### with 
<img width="1000" alt="image" src="https://github.com/user-attachments/assets/73c6b5a7-b577-4dc1-844c-99470fd96fec" />


Resolves: [STT-1363]


[STT-1363]: https://sofab.atlassian.net/browse/STT-1363?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

eos87 committed:
Sanitize assignment description HTML to plain text

Updated `DescriptionTextComponent` to convert HTML in assignment descriptions to plain text using `stringUtils.convertHtmlToPlainText`. This prevents rendering raw HTML.

STT-1363

eos87 committed:
Add selector for assignment coverage profile

Introduces `getCurrentAssignmentCoverageProfile` selector to retrieve the coverage profile used when creating an assignment, including user modifications. Updates AssignmentPreview and AssignmentPreviewContainer to use this profile, and adds tests for the new selector.

STT-1363

tomaskikutis commented on client/components/fields/preview/base/PreviewHoc.tsx:
```diff
@@ -63,7 +63,7 @@ export function previewHoc<S = {}>(
                     {...props}
                     {...this.props}
-                    schema={this.props.schema?.[field]}
+                    schema={this.props.schema}
```
Are you sure this is correct? Not saying it's not, but you're replacing field specific schema item with the entire schema that is field agnostic.

eos87 replied:
I have double checked and I was right and wrong at the same time. There are instances where the schema comes ready for the given field and case where the entire schema is passed. I'll adjust it and restore this piece to how it was.

eos87 replied:
on the other hand, the type seems wrong as it's already for a single field schema not the entire map https://github.com/superdesk/superdesk-planning/blob/40ca0889390a36139444af11f44bb905586414dd/client/components/fields/preview/base/PreviewHoc.tsx#L21

tomaskikutis reviewed (approved):
Avoid performing boolean operations on non-boolean values. e.g. using exclamation mark operator or using a string/number/undefined inside an `if` condition.

Type coercion makes it work in js, but also can lead to mistakes so it's better to avoid it.

<hr>

Regarding tests, I wouldn't add that many tests for that selector. The code is quite straigh-forward there. I'd better test the entire preview component and inspect output to verify that HTML tags were stripped out.

eos87 committed:
Pass field-specific schema to preview components

Updated AssignmentPreview and CoveragePreview to pass the appropriate schema object to preview components. Modified PreviewHoc to accept a schema object mapping and extract the field-specific schema before passing it to the wrapped component.

STT-1363

eos87 committed:
Use explicit boolean checks in `getCurrentAssignmentCoverageProfile`

STT-1363

eos87 commented:
Hey @tomaskikutis, I've implemented your suggestions. Can you please take another look when you get the time? Thanks

eos87 committed:
Remove redundant tests for getCurrentAssignmentCoverageProfile

STT-1363
